### DISCUSSION TITLE
Building bridges

### DISCUSSION THREAD

**Discussion by appenz** [PrismarineJS/mineflayer#2146] [Q&A]
I am trying to build a bot that can bridge over an empty space. The challenge is that the bot needs to move to the very edge and place the block looking behind itself at the side of the block it is standing on, to place the next block. What I tried so far:
- Walking on top of the block via pathfinder at the edge and placing the block fails. This is expected as the bot can't look at the side of the block from its normal position. 
- Moving the bot to the very edge of the block by looking in the right direction, sneaking, and moving forward. This gets the bot into a position where if I am in the same position with a normal client can place the block. However placing the block with placeBlock() fails and it looks like the bot is looking too high when trying to place it. I tried placing while sneaking and while standing with identical results.
- I tried using lookAt() to look at the right position but this didn't work. Looking at the code, the placeBlock() will change the position that is looked at so this is not surprising.
- Tried placeBlockWithOptions but looks like it's internal only (I am using it via the Python bridge)
- Tried with generic_place() and forcelook='ignore' but it still seems to look in the wrong place

Before I dive deeper into the code, has anyone else managed to do this successfully? Any pointers would be super appreciated!

   Guido

**Comment by IHasBone** [comment]
maybe use mineflayer-scaffold?
